routes/admins/admin_codes.js

```diff
@@ -0,0 +1,70 @@
+const crypto = require('crypto');
+
+const express = require('express');
+const { pool } = require('../../server/db');
+// const { keysToCamel } = require('../utils');
+
+const adminCodeRouter = express();
+
+adminCodeRouter.get('/', async (req, res) => {
+  try {
+    const {
+      cookies: { userId },
+    } = req;
+    const user = await pool.query('SELECT DISTINCT code FROM admin_codes WHERE user_id = $1', [
+      userId,
+    ]);
+    res.status(200).send(user.rows);
+  } catch (err) {
+    res.status(400).send(err.message);
+  }
+});
+
+adminCodeRouter.get('/code/:adminCode', async (req, res) => {
+  try {
+    const { adminCode } = req.params;
+    const user = await pool.query('SELECT * FROM admin_codes WHERE admin_codes.code = $1', [
+      adminCode,
+    ]);
+    res.status(200).send(user.rows);
+  } catch (err) {
+    res.status(400).send(err.message);
+  }
+});
+
+adminCodeRouter.post('/', async (req, res) => {
+  try {
+    const {
+      cookies: { userId },
+    } = req;
+    const adminCode = crypto.randomBytes(4).toString('hex');
+    const data = await pool.query(
+      'INSERT INTO admin_codes(user_id, code) VALUES ($1, $2) RETURNING *',
+      [userId, adminCode],
+    );
+    res.status(200).send(data.rows);
+  } catch (err) {
+    res.status(400).send(err.message);
+  }
+});
+
+// delete a user
+adminCodeRouter.delete('/:adminCode', async (req, res) => {
+  try {
+    const {
+      cookies: { userId },
+    } = req;
+    const { adminCode } = req.params;
+    const deletedCode = await pool.query(
+      `DELETE FROM admin_codes
+      WHERE code = $1 AND user_id = $2

```

Comment:
don't think we need to specify the user id to be able to remove an admin code

Reply:
ur absolutely right, this is why onSubmit for CreateAccount failed to delete the code